lib/ukallocbbuddy/bbuddy.c

```diff
@@ -312,6 +388,29 @@ static void bbuddy_pfree(struct uk_alloc *a, void *obj, unsigned long num_pages)
 	/* First free the chunk */
 	map_free(b, (uintptr_t)obj, 1UL << order);
 
+	/* Add the free chunk to the balloon */
```

Comment:
I really don't like having this within the allocator logic in `bbuddy_pfree`. I think that it should be in a separate function especially since there's a lot of duplicate code with other balloon-related stuff in other functions.